File: Groove.SP/Groove.SP.Core/Entities/BillOfLadingModel.cs
using System;
using System.Collections.Generic;

namespace Groove.SP.Core.Entities
{
    public class BillOfLadingModel : Entity
    {
        public long Id { get; set; }

        public string BillOfLadingNo { get; set; }

        /// <summary>
        /// OrganizationID
        /// </summary>
        public long? ExecutionAgentId { get; set; }

        public string BillOfLadingType { get; set; }

        public string MainCarrier { get; set; }

        public string MainVessel { get; set; }

        public decimal TotalGrossWeight { get; set; }

        public string TotalGrossWeightUOM { get; set; }

        public decimal TotalNetWeight { get; set; }

        public string TotalNetWeightUOM { get; set; }

        public decimal TotalPackage { get; set; }

        public string TotalPackageUOM { get; set; }

        public decimal TotalVolume { get; set; }

        public string TotalVolumeUOM { get; set; }

        public string JobNumber { get; set; }

        public DateTime IssueDate { get; set; }

        public string ModeOfTransport { get; set; }

        public string ShipFrom { get; set; }

        public DateTime ShipFromETDDate { get; set; }

        public string ShipTo { get; set; }

        public DateTime ShipToETADate { get; set; }

        public string Movement { get; set; }

        public string Incoterm { get; set; }

        public virtual ICollection<BillOfLadingContactModel> Contacts { get; set; }

        public virtual ICollection<ShipmentBillOfLadingModel> ShipmentBillOfLadings { get; set; }

        public virtual ICollection<BillOfLadingItineraryModel> BillOfLadingItineraries { get; set; }

        public virtual ICollection<ConsignmentModel> Consignments { get; set; }

        /// <summary>
        /// To define linkage between bill of lading and master bill of lading
        /// </summary>
        public virtual ICollection<BillOfLadingShipmentLoadModel> BillOfLadingShipmentLoads { get; set; }

        public virtual ICollection<BillOfLadingConsignmentModel> BillOfLadingConsignments { get; set; }
    }
}

File: Groove.SP/groove-sp-web/src/app/features/shipment/bill-of-lading/models/bill-of-lading.model.ts
import { BaseModel } from 'src/app/core/models/base.model';

/**
 * House bill of lading model
 */
export interface BillOfLadingModel extends BaseModel {
    id: number;
    /**
     * House bill of lading number
     */
    billOfLadingNo: string;
    executionAgentId: number | null;
    billOfLadingType: string;
    mainCarrier: string;
    mainVessel: string;
    totalGrossWeight: number | null;
    totalGrossWeightUOM: string;
    totalNetWeight: number | null;
    totalNetWeightUOM: string;
    totalPackage: number | null;
    totalPackageUOM: string;
    totalVolume: number | null;
    totalVolumeUOM: string;
    jobNumber: string;
    issueDate: string;
    modeOfTransport: string;
    shipFrom: string;
    shipFromETDDate: Date;
    shipTo: string;
    shipToETADate: Date;
    movement: string;
    incoterm: string;
    shipper: string;
    originAgent: string;
    destinationAgent: string;
    consignee: string;
    notifyParty: string;
    nominationPrincipal: string;
    contacts: Array<BillOfLadingContactModel>;

    // master bill of lading
    masterBillOfLadingNo: string;
    masterBillOfLadingId?: number;
}

export interface BillOfLadingContactModel extends BaseModel {
    id: number;
    billOfLadingId: number;
    organizationId: number;
    organizationRole: string;
    companyName: string;
    address: string;
    contactName: string;
    contactNumber: string;
    contactEmail: string;
}
